cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceEnsurer.java

```diff
@@ -40,17 +44,21 @@
   private final Service serviceDelegate;
 
   @Inject
-  public DefaultNamespaceEnsurer(final NamespaceAdmin namespaceAdmin) {
+  public DefaultNamespaceEnsurer(CConfiguration cConfiguration, final NamespaceAdmin namespaceAdmin) {
+    final boolean createDefaultNamespace = cConfiguration.getBoolean(Constants.Namespace.CREATE_DEFAULT_NAMESPACE);
```

Comment:
So frustrating to see dependency like this. For now, what I suggest is, you do this in here:

```java
private final Service serviceDelegate;

@Inject
DefaultNamespaceEnsurer(CConfiguration cConfiguration, final NamespaceAdmin namespaceAdmin) {
  if (!cConfiguration.getBoolean(Constants.Namespace.CREATE_DEFAULT_NAMESPACE)) {
    this.serviceDelegate = null;
  } else {
    this.serviceDelegate = new RetryOnStartFailureService(....);
  }
}

@Override
protected void doStart() {
  if (serviceDelegate != null) {
    serviceDelegate.start();
  }
  notifyStarted();
}

// Same goes for doStop(). Otherwise we are starting up a new thread unncessarily
```